commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -572,6 +572,10 @@ public void deleteAssetCategoryCPDefinition(
 	public CPDefinition deleteCPDefinition(CPDefinition cpDefinition)
 		throws PortalException {
 
+		// Commerce product
+
+		cProductLocalService.deleteCProduct(cpDefinition.getCProductId());
```

Comment:
@Commando18 we have to check if this is the unique CPDefinition related to that CProduct before deleting the product (In case versioning is enabled).  Like "cpdefinitioncount > 1"